x-pack/plugin/rollup/src/main/java/org/elasticsearch/xpack/rollup/action/TransportPutRollupJobAction.java

```diff
@@ -176,6 +178,21 @@ static void updateMapping(RollupJob job, ActionListener<PutRollupJobAction.Respo
             }
 
             Map<String, Object> rollupMeta = (Map<String, Object>)((Map<String, Object>) m).get(RollupField.ROLLUP_META);
+
+            String stringVersion = (String)((Map<String, Object>) m).get(Rollup.ROLLUP_TEMPLATE_VERSION_FIELD);
+            if (stringVersion == null) {
+                logger.warn("Could not determine version of existing rollup metadata for index [" + indexName + "]");
```

Comment:
Is a warning enough here? I am wondering if when the stringVersion is null: a) will we end up throwing a NPE below such as in the `Version.fromString()` method and b) if we will be able to "do the right thing"?